internal/cri/server/container_create.go

```diff
@@ -362,6 +373,106 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 	return &runtime.CreateContainerResponse{ContainerId: id}, nil
 }
 
+func (c *criService) mutateMounts(
+	ctx context.Context,
+	platform imagespec.Platform,
+	extraMounts []*runtime.Mount,
+) error {
+	for _, m := range extraMounts {
+		err := c.mutateOCIBindMount(ctx, platform, m)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *criService) mutateOCIBindMount(
+	ctx context.Context,
+	platform imagespec.Platform,
+	extraMount *runtime.Mount,
+) (retErr error) {
+	imageSpec := extraMount.GetImage()
+	if imageSpec == nil {
+		return nil
+	}
+
+	ref := imageSpec.GetImage()
+	if ref == "" {
+		return fmt.Errorf("image not specified")
+	}
+
+	image, err := c.LocalResolve(ref)
+	if err != nil {
+		return fmt.Errorf("failed to resolve image %q: %w", ref, err)
+	}
+	target := c.getImageVolumeDir(strings.ReplaceAll(image.ID, ":", "/"))
+
+	snapshotter := defaults.DefaultSnapshotter
+
+	ctx, done, err := c.client.WithLease(ctx,
```

Comment:
lease may need to be scoped to the pod?

Reply:
@dmcgowan thoughts?   https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#design-details

Reply:
I agree but this would just be one lease on the same image, or am I thinking about this wrong?